### ISSUE TITLE
Unexpected back behavior in settings

### ISSUE THREAD

**Issue by ezwelty** [falling-fruit/falling-fruit-web#484]
On desktop, it is possible to have the settings tab open and move the map around. If one does so, clicking "<- Back" goes through each of the map views navigated to while settings was open, before closing the settings. I expect it to just close the settings sidebar?

Perhaps related: Clicking on a location while settings is open, then clicking back on location does not go back to settings as one might expect, but to the default map view.

**Comment by badaminatasha** [comment]
I can pick this up. It looks like the back button in the settings page should be using the goToMap function and not goBack.

**Comment by wbazant** [comment]
This issue is not quite solved - #486 adds a 'Go back to map' which is fine for e.g. new location page since adding a new location is only accessible from the map, but we can go to settings from a few other places:
- location tab
- edit review
- edit location

We need the behaviour to be context-specific, and use state in Redux as follows:
```
const {locationId, isBeingEdited: isEditingLocation} = useSelector(state.location)
```
- if there's a location id set but not editing, go to /locations/:locationId
- if there's location id and is editing, go to /locations/:locationId
```
const {reviewId} = useSelector(state.review)
```
- if there's a reviewId set, go to /reviews/:reviewId

Otherwise go back to the map.
